stream-chat-android-ui-components-sample/src/main/kotlin/io/getstream/chat/ui/sample/feature/channel/add/group/select_name/AddGroupChannelSelectNameFragment.kt

```diff
@@ -0,0 +1,108 @@
+        setHasOptionsMenu(true)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        _binding = FragmentAddGroupChannelSelectNameBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+        inflater.inflate(R.menu.menu_add_group_channel_select_name, menu)
+    }
+
+    override fun onPrepareOptionsMenu(menu: Menu) {
+        menu.findItem(R.id.createGroupChannelButton).actionView.findViewById<View>(R.id.addChannelButton).apply {
+            isEnabled = isCreateButtonEnabled
+            setOnClickListener {
+                sharedMembersViewModel.members.value?.let { members ->
+                    viewModelSelect.onEvent(
+                        AddGroupChannelSelectNameViewModel.Event.CreateChannel(
+                            name = binding.nameEditText.text.toString(),
+                            members = members
+                        )
+                    )
+                }
+            }
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initToolbar(binding.toolbar)
+        binding.membersRecyclerView.adapter = adapter
+        adapter.deleteMemberClickListener = AddGroupChannelSelectNameMembersAdapter.DeleteMemberClickListener {
+            sharedMembersViewModel.removeMember(it)
+        }
+        sharedMembersViewModel.members.observe(viewLifecycleOwner) { members ->
+            if (members.isEmpty()) {
+                findNavController().navigateUp()
+            } else {
+                binding.membersTitleTextView.text =
+                    getString(R.string.add_group_channel_select_name_members_title, members.size + 1)
+                adapter.submitList(members)
+            }
+        }
+        binding.nameEditText.doAfterTextChanged {
+            val shouldEnabledCreateButton = it?.toString()?.isNotEmpty() ?: false
+            if (shouldEnabledCreateButton != isCreateButtonEnabled) {
+                isCreateButtonEnabled = shouldEnabledCreateButton
+                requireActivity().invalidateOptionsMenu()
+            }
+        }
+        viewModelSelect.state.observe(viewLifecycleOwner) { state ->
+            when (state) {
+                AddGroupChannelSelectNameViewModel.State.Loading -> {
+                    binding.membersRecyclerView.isVisible = false
+                    binding.progressBar.isVisible = true
+                }
+                is AddGroupChannelSelectNameViewModel.State.NavigateToChannel -> {
+                    sharedMembersViewModel.setMembers(emptyList())
+                    findNavController().navigateSafely(
+                        AddGroupChannelSelectNameFragmentDirections.actionOpenChat(state.cid, null)
+                    )
```

Comment:
I get navigated back to the previous screen ("Add Group Members") instead of the new chat